lib/blpapi-wrapper.ts

```diff
@@ -336,8 +354,17 @@ export class Session extends events.EventEmitter {
                 // XXX: yes, this is a side-effect of map, but it is needed for performance reasons
                 //      until ES6 Map is available
                 this.subscriptions[cid] = s;
-                this.listen('MarketDataEvents', cid, (m: any): void => {
-                    s.emit('data', m.data, s);
+
+                // Lookup by service the events we need to listen to.
+                if (!(s.serviceUri in SERVICE_TO_SUBSCRIPTION_EVENTS_MAP)) {
+                    throw new Error('Unknown service: ' + s.serviceUri);
+                }
+                var events: string[] = SERVICE_TO_SUBSCRIPTION_EVENTS_MAP[s.serviceUri];
```

Comment:
I don't think there is a need to capture the `events` variable.  This could all be done functionally in-line.